plugins/lazy.js

```diff
@@ -3,35 +3,21 @@ import { joinURLPathSegments } from '../lib/utils.js';
 
 export default fp(
   async (fastify, { enabled, prefix = '/', base, development = false }) => {
+    // @ts-ignore
+    if (!fastify.scriptsList) fastify.decorate('scriptsList', []);
+
     // inject live reload when in dev mode
     if (enabled) {
-      fastify.addHook(
-        'onSend',
-        (request, reply, /** @type {string} */ payload, done) => {
-          let newPayload = payload;
-          const contentType = reply.getHeader('content-type') || '';
-          if (typeof contentType === 'string') {
-            // only inject lazy if the content type is html
-            if (contentType.includes('html')) {
-              // if there is a document, inject before closing body
-              const url = development
-                ? joinURLPathSegments(prefix, `/_/dynamic/files/lazy.js`)
-                : joinURLPathSegments(base, `/client/lazy.js`);
+      const url = development
+        ? joinURLPathSegments(prefix, `/_/dynamic/files/lazy.js`)
+        : joinURLPathSegments(base, `/client/lazy.js`);
 
-              if (payload.includes('</body>')) {
-                newPayload = payload.replace(
-                  '</body>',
-                  `<script type="module">addEventListener('load',()=>import('${url}'))</script></body>`,
-                );
-              } else {
-                // if no document, inject at the end of the payload
-                newPayload = `${payload}<script type="module">addEventListener('load',()=>import('${url}'))</script>`;
-              }
-            }
-          }
-          done(null, newPayload);
-        },
-      );
+      // @ts-ignore
+      fastify.scriptsList.push({
```

Comment:
these scriptsList.push calls are used to gather up assets for later use to do a podlet.js() call in the podlet plugin